mls-rs-uniffi/src/lib.rs

```diff
@@ -0,0 +1,524 @@
+    ) -> Result<Message, Error> {
+        let key_package_message = arc_unwrap_or_clone(key_package_message);
+        let key_package = key_package_message
+            .inner
+            .into_key_package()
+            .ok_or(MlsError::UnexpectedMessageType)?;
+        let identifier = signing_identity_to_identifier(key_package.signing_identity())?;
+        let mut group = self.inner().await;
+        let member = group.member_with_identity(&identifier).await?;
+        let mls_message = group.propose_remove(member.index, Vec::new()).await?;
+        Ok(mls_message.into())
+    }
+
+    /// Encrypt an application message using the current group state.
+    pub async fn encrypt_application_message(&self, message: &[u8]) -> Result<Message, Error> {
+        let mut group = self.inner().await;
+        let mls_message = group
+            .encrypt_application_message(message, Vec::new())
+            .await?;
+        Ok(mls_message.into())
+    }
+
+    /// Process an inbound message for this group.
+    pub async fn process_incoming_message(
+        &self,
+        message: Arc<Message>,
+    ) -> Result<ReceivedMessage, Error> {
+        let message = arc_unwrap_or_clone(message);
+        let mut group = self.inner().await;
+        match group.process_incoming_message(message.inner).await? {
+            group::ReceivedMessage::ApplicationMessage(application_message) => {
+                let sender = Arc::new(
+                    self.index_to_identity(application_message.sender_index)
+                        .await?
+                        .into(),
+                );
+                let data = application_message.authenticated_data;
+                Ok(ReceivedMessage::ApplicationMessage { sender, data })
+            }
+            group::ReceivedMessage::Commit(commit_message) => {
+                let committer = Arc::new(
+                    self.index_to_identity(commit_message.committer)
+                        .await?
+                        .into(),
+                );
+                Ok(ReceivedMessage::Commit { committer })
+            }
+            group::ReceivedMessage::Proposal(proposal_message) => {
+                let sender = match proposal_message.sender {
+                    mls_rs::group::ProposalSender::Member(index) => {
+                        Arc::new(self.index_to_identity(index).await?.into())
+                    }
+                    _ => todo!("External and NewMember proposal senders are not supported"),
+                };
+                let proposal = Arc::new(proposal_message.proposal.into());
+                Ok(ReceivedMessage::Proposal { sender, proposal })
+            }
+            // TODO: group::ReceivedMessage::GroupInfo does not have any
+            // public methods (unless the "ffi" Cargo feature is set).
+            // So perhaps we don't need it?
+            group::ReceivedMessage::GroupInfo(_) => Ok(ReceivedMessage::GroupInfo),
```

Comment:
It has methods in [here](https://github.com/awslabs/mls-rs/blob/main/mls-rs/src/group/group_info.rs#L29) but the type GroupInfo is private due to a forgotten export which ought to be fixed.

Reply:
I thought the methods there were only meant to be used with safer-ffi?

I'll just leave this open for now, then we can return to it when we have more of the API built out.

Reply:
I hope the TODO can do for now — we'll find this quickly when we start building on top of the API here.